app/models/manageiq/providers/openshift/inventory/collector/container_manager.rb

```diff
@@ -1,57 +1,66 @@
 class ManageIQ::Providers::Openshift::Inventory::Collector::ContainerManager < ManageIQ::Providers::Kubernetes::Inventory::Collector::ContainerManager
-  attr_reader :version
+  def clusterversion
+    @clusterversion ||= begin
+      openshift_connection("config.openshift.io/v1").get_cluster_version("version") if openshift_version == "v4"
```

Comment:
What about v3?

Reply:
create_rest_client is undocumented.  Kubeclient might use another http library in future.

The right way is to implement https://github.com/abonas/kubeclient/issues/389, hint hint :wink: